seedbuilder/areas.ori

```diff
@@ -1479,79 +1479,90 @@ home: OuterSwampUpperArea 637 -54
 		casual DoubleJump
 		standard Free
 	conn: OuterSwampLowerArea
+	conn: OuterSwampSwarmLedge
+		standard Free
 	conn: SwampTeleporter
 home: OuterSwampAbilityCellNook 692 -83
 	-- anchor: on the ledge near the ability cell
 	pickup: OuterSwampAbilityCell
 	conn: InnerSwampSkyArea
 		glitched Dash
-home: OuterSwampLowerArea 528 -120
+home: OuterSwampLowerArea 650 -100
 	-- anchor: on the ground among the mortar worms
-	pickup: OuterSwampStompExp
+	conn: OuterSwampSwarmLedge
 		casual WallJump
 		casual Climb
 		casual Bash
 		casual ChargeJump
 		casual DoubleJump
 		casual Glide Wind
-		expert ChargeDash
-	pickup: OuterSwampHealthCell
-		casual WallJump DoubleJump
-		casual ChargeJump
+		expert ChargeDash Energy=1
+	conn: OuterSwampAbilityCellNook
 		casual Bash Grenade
-		standard Climb DoubleJump
-		expert ChargeDash
-		master Bash
-		master DoubleJump Glide
-		master TripleJump
-	conn: OuterSwampMortarAbilityCellLedge
-		casual Bash
-		expert ChargeDash
-		master WallJump TripleJump
+		casual Glide Wind
+		standard Bash
+		standard WallJump
+		standard Climb
 	conn: OuterSwampUpperArea
 		casual WallJump DoubleJump
 		casual Climb DoubleJump
 		casual ChargeJump
 		casual Bash Grenade
 		casual Glide Wind
-		-- I still don't understand why people think this jump is hard
 		expert WallJump
 		expert Climb
-		expert ChargeDash
-	conn: OuterSwampAbilityCellNook
+		expert ChargeDash Energy=1
+home: OuterSwampSwarmLedge 600 -90
+	-- anchor: where the swarm spawns
+	pickup: OuterSwampStompExp
+		casual Water
+		standard Free
+	pickup: OuterSwampHealthCell
+		casual WallJump DoubleJump
+		casual ChargeJump
 		casual Bash Grenade
-		casual Glide Wind
-		standard Bash
-		standard WallJump
-		standard Climb
+		standard Climb DoubleJump Glide
+		expert Climb DoubleJump
+		expert ChargeDash Energy=1
+		master Bash
+		master DoubleJump Glide
+		master TripleJump
+	pickup: OuterSwampMortarPlant
+		standard Grenade
 	conn: OuterSwampMortarPlantAccess
 		casual WallJump DoubleJump
 		casual Climb DoubleJump
+		casual WallJump Glide
+		casual Climb Glide
 		casual Bash
 		casual ChargeJump
-	conn: SwampEntryArea
-		casual WallJump
-		casual Climb
-		casual Bash
+		standard DoubleJump
+		standard Glide
+		standard AirDash
+		expert Dash
+	conn: OuterSwampMortarAbilityCellLedge
+		-- these paths assume that you solve the puzzle
 		casual ChargeJump
```

Comment:
This is not a casual path, it turns out, if you assume the puzzle has been completed. You _can_ get back up to the ability cell without the log platforms, but I'd consider it roughly expert-level precision. "Just leave the logs there" is a perfectly reasonable approach for standard, but for casual let's add double jump. So the suggested changes would be:
```
casual ChargeJump DoubleJump
standard ChargeJump
```